### PR TITLE
Kippo Graph Support

### PR DESCRIPTION
**Pull request by aabed**
That's my initial implantation for kippo graph into MHN using pygal

### CODE REVIEW

**Comment by aabed** [comment]
Image
Image

**Comment by jt6211** [comment]
@aabed thanks for working on this.  The code looks good to me, but I want to try it out before merging.  I am hoping to get some time this week to pull this code down and try it out.  Thanks.

**Comment by aabed** [comment]
@jt6211 Awesome, thank you
Please try it before merging, I am not so sure what I've done ;) lol
I want to mention that this is not the final work, I will add more graphs
The next stage will be a more generic graphs that will take the name of the honeypot,sensor_name/ip
and plot the graphs based on that input :)
Hopefully I shall have some time to do so soon

Regards,

**Comment by jt6211** [comment]
`pygal==1.7.0` needs to be added to the `server/requirements.txt` file.

File: server/mhn/templates/base.html [hunk 1]
```diff
@@ -42,7 +42,23 @@
                             <li><a href="{{ url_for('ui.add_sensor') }}">Add sensor</a></li>
                         </ul>
                     </li>
+            
+				
+				
+			 <li class="has-dropdown">
+                        <a href="ui/chart">Kippo Graph</a>
```
Comment (jt6211) [inline server/mhn/templates/base.html]: this needs to be changed ("/" before "ui/"):

```
<a href="/ui/chart">Kippo Graph</a>
```

File: server/mhn/ui/views.py [hunk 2]
```diff
@@ -176,3 +177,79 @@ def forgot_passwd(hashstr):
...
+    print top_attackers    
+    for attacker in top_attackers:
+        bar_chart.add(str(attacker['source_ip']),int(attacker['count']))
+
+    return bar_chart.render_response()
+
+
+
+@ui.route('/chart')
+def chart():
```
Comment (jt6211) [inline server/mhn/ui/views.py]: can you remove the commented out code?

**Comment by jt6211** [comment]
These changes are also required.  Without them, this page breaks if MHN is deployed over HTTPS:

```
diff --git a/server/mhn/templates/base.html b/server/mhn/templates/base.html
index 84160b2..8e540b8 100644
--- a/server/mhn/templates/base.html
+++ b/server/mhn/templates/base.html
@@ -46,7 +46,7 @@
                                
                                
                         <li class="has-dropdown">
-                        <a href="ui/chart">Kippo Graph</a>
+                        <a href="/ui/chart">Kippo Graph</a>
                         <ul class="dropdown">
                             <li><a href="/image/top_passwords.svg">Top Passwords</a></li>
                             <li><a href="/image/top_users.svg">Top Users</a></li>
diff --git a/server/mhn/ui/views.py b/server/mhn/ui/views.py
index c23e1a1..22e14a2 100644
--- a/server/mhn/ui/views.py
+++ b/server/mhn/ui/views.py
@@ -21,6 +21,12 @@ from mhn.common.clio import Clio
 ui = Blueprint('ui', __name__, url_prefix='/ui')
 from mhn import mhn as app
 
+PYGAL_CONFIG = pygal.config.Config()
+PYGAL_CONFIG.js = (
+    'https://kozea.github.io/pygal.js/javascripts/svg.jquery.js',
+    'https://kozea.github.io/pygal.js/javascripts/pygal-tooltips.js',
+)
+
 @app.template_filter()
 def number_format(value):
     return '{:,d}'.format(value)
@@ -185,7 +191,7 @@ def reset_passwd():
 def graph_passwords():
     clio=Clio()
     
-    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True)
+    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True, config=PYGAL_CONFIG)
     bar_chart.title = "Top 10 Passwords"
     clio=Clio()
     top_passwords =clio.hpfeed.count_passwords(clio.hpfeed.get_payloads({'limit':10000},{"channel":"kippo.sessions"})[2])
@@ -199,7 +205,7 @@ def graph_passwords():
 def graph_users():
     clio=Clio()
     
-    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True)
+    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True, config=PYGAL_CONFIG)
     bar_chart.title = "Top 10 Users"
     clio=Clio()
     top_users =clio.hpfeed.count_users(clio.hpfeed.get_payloads({'limit':10000},{"channel":"kippo.sessions"})[2])
@@ -214,7 +220,7 @@ def graph_users():
 def graph_combos():
     clio=Clio()
     
-    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True)
+    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True, config=PYGAL_CONFIG)
     bar_chart.title = "Top 10 Combos"
     clio=Clio()
     top_combos =clio.hpfeed.count_combos(clio.hpfeed.get_payloads({'limit':10000},{"channel":"kippo.sessions"})[2])
@@ -231,7 +237,7 @@ def graph_combos():
 def graph_top_attackers():
     clio=Clio()
     
-    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True)
+    bar_chart = pygal.Bar(style=LightColorizedStyle,show_x_labels=True, config=PYGAL_CONFIG)
     bar_chart.title = "Top 10 Successful Attackers"
     clio=Clio()
     top_attackers =clio.session._tops('source_ip',10,honeypot='kippo')
```

**Comment by aabed** [comment]
I was working on the fixes and, while you were too :)
Thanks @jt6211

**Pull request merged**